subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -221,6 +233,42 @@ private static void validatePackageName(String packageName) {
         }
     }
 
+    static class JavaVersionSelection {
+        private final JavaLanguageVersion languageVersion;
+
+        JavaVersionSelection(JavaLanguageVersion ver) {
+            languageVersion = ver;
+        }
+
+        @Override
+        public String toString() {
+            return "Java " + languageVersion;
+        }
+
+        public JavaLanguageVersion getLanguageVersion() {
+            return languageVersion;
+        }
+    }
+
+    java.util.Optional<JavaLanguageVersion> getJavaLanguageVersion(boolean useIncubatingAPIs, UserInputHandler inputHandler, Language language) {
+        if (isJvmLanguage(language) && useIncubatingAPIs) {
+            JavaLanguageVersion current = JavaLanguageVersion.of(Jvm.current().getJavaVersion().getMajorVersion());
+            List<JavaVersionSelection> knownVersionSelections = DefaultJavaLanguageVersion.getKnownVersions().map(version -> new JavaVersionSelection(version)).collect(toList());
+            return of(inputHandler.selectOption(
+                "Which version of Java is this Project targeting",
+                knownVersionSelections,
+                new JavaVersionSelection(current)))
+                .map(JavaVersionSelection::getLanguageVersion);
```

Comment:
Do we need to constrain the version here at all? Can't someone provision a build with a higher Java version?

I would change this around to...
- Ask "Which version of Java is this project targeting?"
- Require this answer to be a number greater than whatever the minimum version our provisioning supports.